src/pal/src/map/virtual.cpp

```diff
@@ -884,8 +886,10 @@ static LPVOID VIRTUALReserveMemory(
 
     // First, figure out where we're trying to reserve the memory and
     // how much we need. On most systems, requests to mmap must be
-    // page-aligned and at multiples of the page size.
-    StartBoundary = (UINT_PTR)lpAddress & ~BOUNDARY_64K;
+    // page-aligned and at multiples of the page size. Unlike on Windows, on
+    // Unix, the allocation granularity is the page size, so the start boundary
+    // and memory size to reserve are not aligned to 64 KB.
+    StartBoundary = (UINT_PTR)lpAddress & ~VIRTUAL_PAGE_MASK;
```

Comment:
Ok thanks for the info. I'll need to instead change the executable memory allocator to reserve blocks with 64 KB granularity.